### PR TITLE
Badge, Pill, Swatch components

### LINKED ISSUE #26: New components: Swatch, Pill, Badge
**Props:**
- color
- fontSize
- radii
- ${space} -> padding

**Theme changes:**
- create `const fontSizes = [ 10, 12, 14, 18, 24, 32, 40, 64, 72 ]` _feel free to change these values_

**Component style defaults:**
All components:
font-size = fontSize prop

Pills and Badges:
We should define our padding with `em` values (ie: 0.5em 1.2em) so that when we change the font-size using the `fontSize` prop, the size(padding) dynamically scales.

Swatch:
Define `width` and `height` as `1em` with `padding: 0;`.

**Intentions:** 
- Swatch should be used strictly to display a color. Doesn't accept any children.
- Pill strictly to display a number.
- Badge strictly to display a string.

**Documentation:**
Similar to [evergreen](https://evergreen.segment.com/components/badge-and-pill), let's document all of these components together.

### PR DESCRIPTION
**Pull request by lederer**
## Overview

Add `Badge`, `Pill`, and `Swatch` components.

`Badge` renders its children (assumed, but not enforced, to be a short string) inside a colored tag.

`Pill` composes `Badge` to render a number (supplied via the required `value` prop) with rounded corners. It ignores any children, so should be used with a self-closing tag.

`Swatch` also composes `Badge` to render a square whose background color is supplied via the required `color` prop, which should be a `string` key from the theme's `colors` object. It ignores any children, so should be used with a self-closing tag. Swatch size is controlled via the `fontSize` prop.

I added `space` and `fontSizes` to `defaultTheme.js`:

- `space` conforms to styled-system's expectation of an array of space values for padding and margin. The zeroth value is `0` and all subsequent values are multiples of `0.8rem` (with the assumption of `html { font-size: 62.5%; }` so that `1rem` === `10px` at default zoom). I've found that using multiples of 8 for spacing is easy to remember and to add, multiply, and divide, and it's common in the wild. It also works nicely with styled-system's array approach, where `space[N]` ends up equaling `N * 0.8rem`. Nicely intuitive.

- `fontSizes` conforms to styled-system's expectation of an array of font sizes. I left the redundant `typeSystem` object in there, since some existing, non-styled-system-aware components make use of it. We can retire it once all components are updated to use styled-system.

### Checklist

- [x] Relevant documentation pages have been created or updated
- [ ] Description of PR is in an appropriate section of the changelog and grouped with similar changes if possible

### Demo

127 0 0 1_3030_components_badge-pill-swatch

Closes #26

### CODE REVIEW

File: packages/core/components/badge/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  ...color.PropTypes,
+  ...fontSize.PropTypes,
+  ...borderRadius.PropTypes
+};
+
+Badge.defaultProps = {
+  mx: 1,
+  my: "auto",
+  px: "0.4em",
+  py: "0.24em",
```
Comment (designmatty) [inline packages/core/components/badge/index.js]: I just ran into this issue myself, but when you define defaultProps for `mx`, `my`, `px`, `py` setting overrides with `m` or `p` no longer works correctly. This happened with the Icon component. I think it's a bug with styled-system.

The work around is to instead set defaults for `m` and `p`:
```
m: [1, "auto"],
p: ["0.4em", "0.24em"],
```

File: packages/core/components/pill/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+Pill.propTypes = {
+  ...Badge.propTypes,
+  value: PropTypes.number.isRequired
+};
+
+Pill.defaultProps = {
+  ...Badge.defaultProps,
+  px: 2,
+  py: 1
```
Comment (designmatty) [inline packages/core/components/pill/index.js]: Same comment as above ^^ for `px` and `py`

**Review by designmatty — CHANGES_REQUESTED** [review]
A couple small changes.

File: packages/core/components/badge/index.js (hunk 1 above)
Comment (lederer) [reply packages/core/components/badge/index.js]: Doh. I ran into this on a previous PR and solved it similarly, then somehow forgot all about it and fell into the same trap here.

**Comment by alkamin** [comment]
What do people think about providing an option to have a circular swatch?

**Comment by lederer** [comment]
Ooh. I like that. Strictly speaking they can do that by passing in a `borderRadius` prop with a high pixel value. But I like the idea of also supporting a boolean `circular` prop. Will do that.

**Comment by alkamin** [comment]
It may be worth leveraging a constant...I think there is a shape one.

**Comment by lederer** [comment]
Even better, since then it can support all those shapes. Nice.

**Comment by lederer** [comment]
Added `shape` prop to `Swatch`

image

**Comment by lederer** [comment]
@designmatty This is ready for another look

**Comment by lederer** [comment]
Ready for another look.

Updates:
- Per discussion with @designmatty, default values for props handled by styled-system are now set within styled-component definitions (typically via `themeGet()`) instead of in `defaultProps`
- Swatch now supports the `borderRadius` prop. 
  - For technical reasons, it will override the `shape` prop if both are passed.
- Swatch size is now controlled via the `size` prop.
  - (Technically `fontSize` also works, because `width` and `height` default to `1em`). 
  - `size` accepts only valid CSS `width, height` values. There is currently no corresponding theme object or array.

**Review by designmatty — APPROVED** [review]

**Pull request merged**